grid_generation/Gen_CICE_grid.py

```diff
@@ -0,0 +1,96 @@
+"""
+Script: generate_cice_grid.py
+
+Description: 
+This script generates a CICE grid from the MOM super grid provided in the input NetCDF file.
+
+Contact: 
+Name: Ezhilsabareesh Kannadasan
+
+Usage:
+python generate_cice_grid.py <input_superGridPath> <output_file>
+- input_superGridPath: Path to the MOM super grid NetCDF file.
+- output_file: Path to the output CICE grid NetCDF file.
+"""
+import numpy as np
+import xarray as xr
+from netCDF4 import Dataset
+import sys
+
+def generate_cice_grid(in_superGridPath, output_file):
+    # Read input files
+    in_superGridFile = xr.open_dataset(in_superGridPath)
+
+    # Constants
+    ULAT = np.deg2rad(in_superGridFile['y'][2::2, 2::2])
+    ULON = np.deg2rad(in_superGridFile['x'][2::2, 2::2])
+    TLAT = np.deg2rad(in_superGridFile['y'][1::2, 1::2])
+    TLON = np.deg2rad(in_superGridFile['x'][1::2, 1::2])
+
+    # MPI
+    HTN = in_superGridFile['dx'] * 100.0  # convert to cm
+    HTN = HTN[1::2, ::2] + HTN[1::2, 1::2]
+    HTE = in_superGridFile['dy'] * 100.0  # convert to cm
+    HTE = HTE[::2, 1::2] + HTE[1::2, 1::2]
+
+    ANGLE = np.deg2rad(in_superGridFile['angle_dx'][1::2, 1::2])
+
+    # Close input files
+    in_superGridFile.close()
+
+    # Create a new NetCDF file
+    nc = Dataset(output_file, 'w', format='NETCDF4')
+
+    # Define dimensions
+    ny, nx = ULAT.shape
+    nc.createDimension('ny', ny)
+    nc.createDimension('nx', nx)
+
+    # Define variables
+    ulat = nc.createVariable('ulat', 'f8', ('ny', 'nx'))
+    ulon = nc.createVariable('ulon', 'f8', ('ny', 'nx'))
+    tlat = nc.createVariable('tlat', 'f8', ('ny', 'nx'))
+    tlon = nc.createVariable('tlon', 'f8', ('ny', 'nx'))
+    htn = nc.createVariable('htn', 'f8', ('ny', 'nx'))
+    hte = nc.createVariable('hte', 'f8', ('ny', 'nx'))
+    angle = nc.createVariable('angle', 'f8', ('ny', 'nx'))
+
+    # Add attributes
+    ulat.units = "radians"
+    ulat.title = "Latitude of U points"
+    ulon.units = "radians"
+    ulon.title = "Longitude of U points"
+    tlat.units = "radians"
+    tlat.title = "Latitude of T points"
+    tlon.units = "radians"
+    tlon.title = "Longitude of T points"
+    htn.units = "cm"
+    htn.title = "Width of T cells on North side."
```

Comment:
```suggestion
    htn.title = "Width of T cells on N side."
```
(to match existing grid.nc)